digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java

```diff
@@ -1184,7 +1184,11 @@ public int trySetChildrenBlockedToReadyOrShortCircuitPlannedOrCanceled(long task
                     " when task_type = " + TaskType.GROUPING_ONLY + " then " + TaskStateCode.PLANNED_CODE +
                     " when " + bitAnd("state_flags", Integer.toString(TaskStateFlags.CANCEL_REQUESTED)) + " != 0 then " + TaskStateCode.CANCELED_CODE +
                     " else " + TaskStateCode.READY_CODE +
-                    " end" +
+                    " end, " +
+                    " started_at = case" +
+                    // Update `started_at` only when it's a group task and the column is null.
+                    // Non group tasks' `started_at` are updated by TaskControlStore.setStartedState.
+                    " when task_type = " + TaskType.GROUPING_ONLY + " then coalesce(started_at, now()) else started_at end" +
```

Comment:
[IMO] It would be better if you left a note about the cases that `started_at` is NULL and not NULL in this comment.